api/src/test/scala/za/co/absa/loginsvc/rest/provider/ad/ldap/ActiveDirectoryLDAPAuthenticationProviderTest.scala

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright 2023 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.loginsvc.rest.provider.ad.ldap
+
+import org.mockito.ArgumentMatchers.any
+import org.mockito.Mockito.{mock, times, verify, when}
+import org.mockito.invocation.InvocationOnMock
+import org.mockito.stubbing.Answer
+import org.scalatest.flatspec.AnyFlatSpec
+import org.scalatest.matchers.should.Matchers
+import org.springframework.security.authentication.{
+  AuthenticationProvider,
+  UsernamePasswordAuthenticationToken}
+import org.springframework.security.core.Authentication
+import org.springframework.security.core.authority.SimpleGrantedAuthority
+import org.springframework.security.core.userdetails.{User, UserDetails}
+import za.co.absa.loginsvc.model.{User => CustomUser}
+import za.co.absa.loginsvc.rest.config.auth.{
+  ActiveDirectoryLDAPConfig,
+  LdapRetryConfig,
+  LdapUserCredentialsConfig,
+  ServiceAccountConfig}
+
+import java.util.Collections
+
+class ActiveDirectoryLDAPAuthenticationProviderTest extends AnyFlatSpec with Matchers {
+
+  private val integratedCfg = LdapUserCredentialsConfig("svc-ldap", "password")
+  private val serviceAccountCfg = ServiceAccountConfig(
+    "CN=%s,OU=Users,OU=Accounts,DC=domain,DC=subdomain,DC=com",
+    Option(integratedCfg),
+    None)
+  private val ldapCfg = ActiveDirectoryLDAPConfig(
+    "some.domain.com",
+    "ldaps://some.domain.com:636/",
+    "SomeAccount",
+    1,
+    serviceAccountCfg,
+    None,
+    None,
+    None)
+
+  private val principal = CustomUser("user", List("ROLE_USER"), Map("testAtt" -> None))
+  private val mockAuthentication = mock(classOf[Authentication])
+
+  private class TestActiveDirectoryLDAPAuthenticationProvider(config: ActiveDirectoryLDAPConfig, mockAuthenticationProvider: AuthenticationProvider)
+    extends ActiveDirectoryLDAPAuthenticationProvider(config) {
+    private var counter: Int = 1
+    override private[ldap] def createAuthenticationProvider = mockAuthenticationProvider
+    when(mockAuthenticationProvider.authenticate(any[Authentication])).thenAnswer((_: InvocationOnMock) => {
+      val userDetails: UserDetails = UserDetailsWithExtras(
+        new User("user", "password", Collections.singleton(
+          new SimpleGrantedAuthority("ROLE_USER"))), Map("testAtt" -> None))
```

Comment:
```suggestion
      val userDetails: UserDetails = UserDetailsWithExtras(
        new User("user", "password", Collections.singleton(new SimpleGrantedAuthority("ROLE_USER"))),
        Map("testAtt" -> None)
      )
```

formatting suggestion